### ISSUE TITLE
Make `git_prompt_info` ignore ignored submodules from `.gitmodules`

### ISSUE THREAD

**Issue by oryband** [robbyrussell/oh-my-zsh#2612]
I have my [dotvim][dotvim] repo be a submodule inside my [dotfiles], and made [git ignore any further commits I make inside `.vim/` using `.gitmodules`][gitmodules]. However, `git_prompt_info` shows my repo is dirty, even though it shouldn't:
```sh
~/Documents/dotfiles(master ✗) git status
On branch master
Your branch is up-to-date with 'origin/master'.

nothing to commit, working directory clean

~/Documents/dotfiles(master ✗) git_prompt_info      
(master ✗)
```
I'm using [simple][simple] theme, but I don't think it matters. The prompt should display `(master ✔)` instead.

[dotvim]: https://github.com/oryband/dotvim
[dotfiles]: https://github.com/oryband/dotfiles
[gitmodules]: https://github.com/oryband/dotfiles/blob/860c0e7a1ab7dfa09e0b00a639fd0fd70d734431/.gitmodules
[simple]: https://github.com/robbyrussell/oh-my-zsh/blob/210c1ef86deed805b3c5b25833d4b128f758b610/themes/simple.zsh-theme

**Comment by oryband** [comment]
@robbyrussell why did you close this?

**Comment by oryband** [comment]
Can I get a response please?

**Comment by ncanceill** [comment]
Hi, sorry, no idea why this was closed. However, no worries needed, this issue should be fixed by #2214, which is still open.

**Comment by teo-sk** [comment]
Hmm, it is still open, and this issue is still relevant.. Or is there any news?
